lib/stdlib/src/proc_lib.erl

```diff
@@ -265,12 +265,12 @@ exit_p(Class, Reason) ->
 	{M,F,A} when is_atom(M), is_atom(F), is_integer(A) ->
 	    MFA = {M,F,make_dummy_args(A, [])},
 	    crash_report(Class, Reason, MFA),
-	    exit(Reason);
+	    erlang:raise(Class, Reason, erlang:get_stacktrace());
```

Comment:
You will want to pass `erlang:get_stacktrace()` as an argument to `exit_p/2` instead of calculating it here. The reason is that if any of the code between calling `exit_p/2` and calling `erlang:get_stacktrace()` uses `catch` or `try`, it can potentially change the value of `erlang:get_stacktrace()`. To avoid such bugs being introduced, I typically call `erlang:get_stacktrace()` as the first thing inside my catch clauses.